scripts/ci/jobs/check-generated.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/env bash
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../../.. && pwd)"
+source "$ROOT/scripts/ci/lib.sh"
+
+set -euo pipefail
+
+go mod tidy
+
+
+# shellcheck disable=SC2016
+echo 'Ensure that generated files are up to date. (If this fails, run `make proto-generated-srcs && make go-generated-srcs` and commit the result.)'
+function generated_files-are-up-to-date() {
+    git ls-files --others --exclude-standard >/tmp/untracked
+    make proto-generated-srcs
+    # Print the timestamp along with each new line of output, so we can track how long each command takes
+    make go-generated-srcs 2>&1 | while IFS= read -r line; do printf '[%s] %s\n' "$(date '+%Y-%m-%d %H:%M:%S')" "$line"; done
+    git diff --exit-code HEAD
+    { git ls-files --others --exclude-standard ; cat /tmp/untracked ; } | sort | uniq -u >/tmp/untracked-new
+
+    store_test_results /tmp/untracked-new untracked-new
+
+    if [[ -s /tmp/untracked-new ]]; then
+        # shellcheck disable=SC2016
+        echo 'Found new untracked files after running `make proto-generated-srcs` and `make go-generated-srcs`. Did you forget to `git add` generated mocks and protos?'
+        cat /tmp/untracked-new
+        exit 1
+    fi
+}
+generated_files-are-up-to-date
```

Comment:
Regarding debugging the failure: I have a suspicion that prow is not sharing all the output before it handles the exit. 
```suggestion
generated_files-are-up-to-date || touch FAIL
[[ ! -f FAIL ]] || { echo "Failed, waiting for prow to wind down"; sleep 300; exit 1; }
```